ambari-server/src/main/java/org/apache/ambari/server/upgrade/UpgradeCatalog271.java

```diff
@@ -0,0 +1,175 @@
+          if (installedServices.contains("RANGER")) {
+            Config rangerEnvConfig = cluster.getDesiredConfigByType("ranger-env");
+            Config rangerAdminSiteConfig = cluster.getDesiredConfigByType("ranger-admin-site");
+            Config rangerUgsyncSiteConfig = cluster.getDesiredConfigByType("ranger-ugsync-site");
+            if (rangerEnvConfig != null) {
+              String rangerAdminLogDir = rangerEnvConfig.getProperties().get("ranger_admin_log_dir");
+              String rangerUsersyncLogDir = rangerEnvConfig.getProperties().get("ranger_usersync_log_dir");
+              if (rangerAdminLogDir != null && rangerAdminSiteConfig != null) {
+                Map<String, String> newProperty = new HashMap<String, String>();
+                newProperty.put("ranger.logs.base.dir", rangerAdminLogDir);
+                updateConfigurationPropertiesForCluster(cluster, "ranger-admin-site", newProperty, true, false);
+              }
+              if (rangerUsersyncLogDir != null && rangerUgsyncSiteConfig != null && rangerUgsyncSiteConfig.getProperties().containsKey("ranger.usersync.logdir")) {
+                Map<String, String> updateProperty = new HashMap<String, String>();
+                updateProperty.put("ranger.usersync.logdir", rangerUsersyncLogDir);
+                updateConfigurationPropertiesForCluster(cluster, "ranger-ugsync-site", updateProperty, true, false);
+              }
+              Set<String> removeProperties = Sets.newHashSet("ranger_admin_log_dir", "ranger_usersync_log_dir");
+              removeConfigurationPropertiesFromCluster(cluster, "ranger-env", removeProperties);
+            }
+          }
+        }
+      }
+    }
+
+  }
+
+  protected void updateRangerKmsDbUrl() throws AmbariException {
+    AmbariManagementController ambariManagementController = injector.getInstance(AmbariManagementController.class);
+    Clusters clusters = ambariManagementController.getClusters();
+    if (clusters != null) {
+      Map<String, Cluster> clusterMap = clusters.getClusters();
+      if (clusterMap != null && !clusterMap.isEmpty()) {
+        for (final Cluster cluster : clusterMap.values()) {
+          Set<String> installedServices = cluster.getServices().keySet();
+          if (installedServices.contains("RANGER_KMS")) {
+            Config rangerKmsPropertiesConfig = cluster.getDesiredConfigByType("kms-properties");
+            Config rangerKmsEnvConfig = cluster.getDesiredConfigByType("kms-env");
+            if (rangerKmsPropertiesConfig != null) {
+              String dbFlavor = rangerKmsPropertiesConfig.getProperties().get("DB_FLAVOR");
+              String dbHost = rangerKmsPropertiesConfig.getProperties().get("db_host");
+              String rangerKmsRootDbUrl = "";
+              if (dbFlavor != null && dbHost != null) {
+                if ("MYSQL".equalsIgnoreCase(dbFlavor)) {
+                  rangerKmsRootDbUrl = "jdbc:mysql://" + dbHost + ":3306";
```

Comment:
Assuming the port? I have seen `ambari.properties` files where this is customized.

Reply:
This is the default configured port for each of the database flavors available to generate the JDBC connection url. A similar default port map configuration is used [here](https://github.com/apache/ambari/blob/trunk/ambari-server/src/main/resources/stacks/HDP/2.3/services/stack_advisor.py#L30).

Reply:
Updated the code to fetch port from existing database url.